geode-core/src/main/java/org/apache/geode/internal/cache/TXRegionLockRequestImpl.java

```diff
@@ -111,38 +116,43 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
       if (cache != null && size > 0) {
         this.r = (LocalRegion) cache.getRegion(this.regionPath);
       }
-      this.entryKeys = readEntryKeySet(size, in);
+      this.entryKeys = readEntryKeyMap(size, in);
```

Comment:
Changing the serialized wire-format from a Set to a Map will cause problems during a rolling upgrade if transactions are in progress. What you need to do is introduce versioned fromData/toData methods. When this object is being serialized for the older version it would still serialize a set; some goes for the code reading it off the wire. You can see an example of this here: PartitionMessage.fromData and here:  PartitionMessage.toData. Note how they compare to Version.GFE_90. You would want to compare to Version.GEODE_1_10_0 (or whatever version this fix makes it into). The code you are modifying will always be the new version. But in toData you need to see if you are serializing it to an older version and if so serialize a Set instead of a Map. In fromData you need to see if you are reading from an older version and if so read a Set and then set the Boolean value on your map to a reasonable default.

Reply:
Thanks for comments.